bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/MagnitudeBand.scala

```diff
@@ -1,15 +1,19 @@
 package edu.gemini.spModel.core
 
-sealed abstract class MagnitudeBand private (val name: String, val wavelengthMidPointNm: Option[Int], val description: Option[String], val defaultSystem: MagnitudeSystem) extends Product with Serializable {
+sealed abstract class MagnitudeBand private (val name: String, val center: Wavelength, val width: Wavelength, val description: Option[String], val defaultSystem: MagnitudeSystem) extends Product with Serializable {
 
-  private def this(name: String, wavelengthMidPointNm: Int) =
-    this(name, Some(wavelengthMidPointNm), None, MagnitudeSystem.VEGA)
+  private def this(name: String, center: Wavelength, width: Wavelength) =
+    this(name, center, width, None, MagnitudeSystem.VEGA)
 
-  private def this(name: String, wavelengthMidPointNm: Int, description: String) =
-    this(name, Some(wavelengthMidPointNm), Some(description), MagnitudeSystem.VEGA)
+  private def this(name: String, center: Wavelength, width: Wavelength, description: String) =
+    this(name, center, width, Some(description), MagnitudeSystem.VEGA)
 
-  private def this(name: String, wavelengthMidPointNm: Int, description: String, defaultMagnitudeSystem: MagnitudeSystem) =
-    this(name, Some(wavelengthMidPointNm), Some(description), defaultMagnitudeSystem)
+  private def this(name: String, center: Wavelength, width: Wavelength, description: String, defaultMagnitudeSystem: MagnitudeSystem) =
+    this(name, center, width, Some(description), defaultMagnitudeSystem)
+
+  val start: Wavelength = center + width / 2
+
+  val end:   Wavelength = center - width / 2
```

Comment:
I'd be ok considering the use of `squants`, I like the idea of having compilation errors when adding unit-incompatible objects. I checked and it seems to depend on `spire` and `json4s`. I think we discussed about using `spire` some time ago?